tests/queries/0_stateless/00240_replace_substring_loop.sql

```diff
@@ -99,3 +99,6 @@ SELECT s, replaceOne(s, '_', 'o') AS a, replaceRegexpOne(s, '_', 'o') AS b, a =
 SELECT s, replaceOne(s, '_', 'o') AS a, replaceRegexpOne(s, '_', 'o') AS b, a = b FROM (SELECT arrayJoin(['__.__', '.__']) AS s);
 SELECT s, replaceOne(s, '_', 'o') AS a, replaceRegexpOne(s, '_', 'o') AS b, a = b FROM (SELECT arrayJoin(['__.__', '__.']) AS s);
 SELECT s, replaceOne(s, '_', 'o') AS a, replaceRegexpOne(s, '_', 'o') AS b, a = b FROM (SELECT arrayJoin(['__.__', '__.__']) AS s);
+SELECT replace('ABCabc', '', 'DEF');
```

Comment:
This is way too dirty. We
- should not have 101 lines with tests where `_` is used as pattern, and then suddenly test empty needles _without explanation or comment_ what is being tested
- and, even worse, _without any connection_ to the test name (we are not looping anything),
- and on top, only test the alias "replace" but not test the actual functions "replaceOne" and "replaceAll".

@zhanglistar Please move l. 102 into a new test 00240_replace_with_empty_needle and address these points.